components/core/src/clp/streaming_archive/Constants.hpp

```diff
@@ -15,6 +14,40 @@ constexpr char cMetadataFileName[] = "metadata";
 constexpr char cMetadataDBFileName[] = "metadata.db";
 constexpr char cSchemaFileName[] = "schema.txt";
 
+namespace version {
```

Comment:
The current convention for constants in this file is to name the namespace as `cNamespace` and then the constants inside as `ConstantXxx`. I know that disagrees with our clang-tidy rules, but in this case, I would stick with the existing conventions to avoid confusion until we refactor this file to address the clang-tidy warnings (especially the refactoring will propagate into a lot of files).

If you really want to namespace the version fields (which is fair), how about cArchiveFormatVersion with internal constants "Default" and "Custom"? "Default" isn't the greatest name, but I can't think of something better.

Reply:
> Also in general it seems like people are recommending to me to move to new clang-tidy even in old files. But maybe this is not our recommendation?

It's our recommendation in other cases. In this case, I'm suggesting an exception until we refactor all other constants / make a more long-term decision about how to refactor them.

> Also, I can just default initialize the custom version to 0 in the metadata class. Then we don't need to specify default or custom here. And can just use cVersion. Let me know

True. I like that idea.